src/citrine/resources/gemd_resource.py

```diff
@@ -0,0 +1,92 @@
+"""Collection class for generic GEMD objects and templates."""
+from typing import Type, Union, Optional, List, Iterator
+from uuid import UUID
+
+from gemd.entity.base_entity import BaseEntity
+from gemd.entity.link_by_uid import LinkByUID
+
+from citrine.resources.data_concepts import DataConcepts, DataConceptsCollection
+from citrine._session import Session
+
+
+class GEMDResourceCollection(DataConceptsCollection[DataConcepts]):
+    """A collection of any kind of GEMD objects/templates."""
+
+    _path_template = 'projects/{project_id}/storables'
+    _dataset_agnostic_path_template = 'projects/{project_id}/storables'
+    _individual_key = None
+    _collection_key = None
+    _resource = DataConcepts
+
+    def __init__(self, project_id: UUID, dataset_id: UUID, session: Session):
+        self.project_id = project_id
+        self.dataset_id = dataset_id
+        self.session = session
+
+    @classmethod
+    def get_type(cls) -> Type[DataConcepts]:
+        """Return the resource type in the collection."""
+        return DataConcepts
+
+    def build(self, data: dict) -> DataConcepts:
+        """
+        Build an arbitary GEMD resource from a serialized dictionary.
+
+        This is an internal method, and should not be called directly by users.
+
+        Parameters
+        ----------
+        data: dict
+            A serialized data model object.
+
+        Returns
+        -------
+        DataConcepts
+            A data model object built from the dictionary.
+
+        """
+        return DataConcepts.build(data)
+
+    def update(self, model: DataConcepts) -> DataConcepts:
+        """To update an arbitrary GEMD resource, please use dataset.update instead."""
+        raise NotImplementedError("To update an arbitary GEMD resource,"
+                                  " please use dataset.update instead.")
```

Comment:
Are we forbidding the query `dataset.storables.update(...)`?  It seems like these should check if `self.dataset_id` is set, and dispatch if possible.  Regardless, I don't think this is a `NotImplementedError` error, this would probably be a `ValueError` since they tried to invoke it with `dataset is None`.

https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
@bfolie Can use your input here. The logic was that since the storables route does not have functionality for register/update/delete, we would prefer the user to use the methods on specific collections or on `dataset`, rather than dispatching to that through `GEMDCollection`. 

We can add that dispatch through the parent dataset, but it seems redundant to me. Think it's worth adding in?

Reply:
If the dataset is not None, then it's a `NotImplementedError`, because we haven't implemented it :p
If the method is otherwise implemented but the dataset is `None`, then we've been using `RuntimeError`.

register/update/delete is a nice-to-have. I was more ambivalent when we last discussed it, but now I think it's worth doing, both because users will expect it and because it should be easy to implement by calling up to the already-implemented `Dataset` method. It's true that the `storables/` route doesn't explicitly implement, say, a POST method, but that's immaterial to a user. As far as they're concerned, their intent is clear, so we should be able to deal with it.